### PR TITLE
[RHDEVDOCS-2879] Document ability to customize available cluster roles for Projects

### PR DESCRIPTION
**Pull request by rishumehra**
**Purpose**:

- Customize and configure the cluster roles for a project.
- YAML editing to add or remove roles to the available cluster roles list.

**PR information**:

- Aligned team: DevConsole
- For branches: OpenShift 4.8 Async, OpenShift 4.9
- JIRA: [RHDEVDOCS-2879](https://issues.redhat.com/browse/RHDEVDOCS-2879)
- Direct link to doc preview: https://deploy-preview-40054--osdocs.netlify.app/openshift-enterprise/latest/applications/projects/working-with-projects.html
- SME and QE Review: @jerolimov
- Peer Review: @missmesss @jc-berger

### CODE REVIEW

File: modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc [hunk 1]
```diff
@@ -0,0 +1,40 @@
+// Module included in the following assemblies:
+//
+// applications/projects/working-with-projects.adoc
+
+[id="odc-customizing-available-cluster-roles-using-developer-perspective_{context}"]
+= Customizing the available cluster roles using the Developer perspective
+
+You can customize and configure an existing project access roles name by editing the YAML code of `console.openshift.io/display-name` of your cluster.
```
Comment (jerolimov) [inline modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc:8]: Small details:
* Does it make sense to note here again that you need to changing the `ClusterRole`? But we are in the right section, so it is maybe obvious.
* You should mention that this is an `annotation`, otherwise the admin could also think you mean a label.

```suggestion
You can customize and configure an existing project access roles name by editing the YAML code, and add an `console.openshift.io/display-name: Display name` annotation of your cluster.
```

File: modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+.Procedure
+To customize and configure the cluster roles of your project:
```
Comment (jerolimov) [inline modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc:11]: Did you think it's clear here that all projects are affected if the user changes the cluster configuration?

```suggestion
To customize and configure the cluster roles of all projects:
```

**Review by christoph-jerolimov — COMMENTED** [review]
Two questions, everything else is fine for me.

File: modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc (hunk 1 above)
Comment (rishumehra) [reply modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc:8]: Got it. This suggestion seems fine I'll rephrase the sentence a bit to incorporate the changes.

File: modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc (hunk 2 above)
Comment (rishumehra) [reply modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc:11]: In my understanding, it was only for the current project. And thanks for clarifying that it is for all the projects. I'll update the sentence.

**Review by rishumehra — COMMENTED** [review]
@jerolimov Thanks for your doubts and suggestions. I've replied to your comments/question, please check. I'll make the changes soon.

File: modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+[id="odc-customizing-available-cluster-roles-using-developer-perspective_{context}"]
+= Customizing the available cluster roles using the Developer perspective
+
+You can customize and configure an existing project access roles name by editing the YAML code, and add an `console.openshift.io/display-name: Display name` annotation to the cluster.
+
+.Procedure
+To customize and configure the cluster roles of all the projects:
+
+. In the *Developer* perspective, navigate to the *Search* view.
```
Comment (jerolimov) [inline modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc:13]: Cluster admins work maybe more time in the admin perspective, we have a search there as well which works similar. Maybe we can just drop the perspective recommendation?

```suggestion
. Navigate to the *Search* view.
```

File: modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+//
+// applications/projects/working-with-projects.adoc
+
+[id="odc-customizing-available-cluster-roles-using-developer-perspective_{context}"]
+= Customizing the available cluster roles using the Developer perspective
+
+You can customize and configure an existing project access roles name by editing the YAML code, and add an `console.openshift.io/display-name: Display name` annotation to the cluster.
+
+.Procedure
+To customize and configure the cluster roles of all the projects:
```
Comment (jerolimov) [inline modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc:11]: I would recommend finetuning this here a little bit. The admin will customize the options in the tab here, not the cluster roles themselves. What about this change?

```suggestion
To customize which cluster roles options are shown for all projects in the *Project Access* tab:
```

File: modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc [hunk 5]
```diff
@@ -0,0 +1,40 @@
+// Module included in the following assemblies:
+//
+// applications/projects/working-with-projects.adoc
+
+[id="odc-customizing-available-cluster-roles-using-developer-perspective_{context}"]
+= Customizing the available cluster roles using the Developer perspective
+
+You can customize and configure an existing project access roles name by editing the YAML code, and add an `console.openshift.io/display-name: Display name` annotation to the cluster.
+
```
Comment (jerolimov) [inline modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc:9]: IMHO: I would recommend to move this to the end of the customization doc here. So giving the reader first an introduction how to configure the available options, and then with this hint you can fine-tune the displayed options in the dropdown. Wdyt?

File: modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc (hunk 3 above)
Comment (rishumehra) [reply modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc:13]: Sounds fair, I can rephrase without using "**Developer** perspective".

**Comment by jerolimov** [comment]
Technically I'm fine with your change. Thanks.

/lgtm

File: modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc (hunk 5 above)
Comment (rishumehra) [reply modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc:9]: Thanks for the suggestion but as per the docs consistency, we define or explain the "what" part with the action item then the proper how-to procedure. This is a kind of template to create content on the docs site.

File: modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc (hunk 4 above)
Comment (rishumehra) [reply modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc:11]: Great suggestion, I am gonna make this change.

**Review by rishumehra — COMMENTED** [review]
@jerolimov Thanks for your review again. I've replied to your comments and will push the suggested changes soon.

File: modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+. Navigate to the *Search* view.
+. In the *Search* view, select the *Resources* drop down and search *Console*.
+. Select the *Console `operator.openshift.io/v1`* to add as a new row in resource list.
++
+.Searching Console resource
+image::odc_cluster_console.png[]
+. Select *cluster* under *Name* list.
+. Navigate to *YAML* tab to view and edit the YAML code.
+. In the YAML code under `spec` add or edit the list of `availableClusterRoles` to configure the project roles.
```
Comment (missmesss) [inline modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc:21]:
```suggestion
. In the YAML code under `spec` add or edit the list of `availableClusterRoles`.
```
If I understand it correctly, this whole procedure is about configuring/customizing project roles, so repeating it inside the procedure isn't necessary maybe?

File: modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc [hunk 7]
```diff
@@ -0,0 +1,40 @@
+// Module included in the following assemblies:
+//
+// applications/projects/working-with-projects.adoc
+
+[id="odc-customizing-available-cluster-roles-using-developer-perspective_{context}"]
+= Customizing the available cluster roles using the Developer perspective
+
+You can customize and configure an existing project access roles name by editing the YAML code, and add an `console.openshift.io/display-name: Display name` annotation to the cluster.
```
[the remark this answers is not in the document]
Comment (rishumehra) [reply modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc:8]: I think the "**access role's name**" is fine here. It is actually the list of roles that can be assigned to a user for a particular cluster.

[the remark this answers is not in the document]
Comment (rishumehra) [reply modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc:8]: @missmesss actually the YAML code resides in the `console.openshift.io/display-name: Display name` file. Where we can add/edit the existing list of user roles of a cluster. And the annotation part is to edit the list in the YAML code.

File: modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+
+.Procedure
+To customize the cluster role options of all projects that are shown in the *Project Access* tab:
```
[the remark this answers is not in the document]
Comment (rishumehra) [reply modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc:11]: If I am not wrong, most of the Openshift documentation follows the "To" convention while starting the procedure section.

File: modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+
+. Navigate to the *Search* view.
+. In the *Search* view, select the *Resources* drop down and search *Console*.
```
[the remark this answers is not in the document]
Comment (rishumehra) [reply modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc:14]: Got it, that's a good point, thanks!

File: modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+. Select the *Console `operator.openshift.io/v1`* to add as a new row in resource list.
```
[the remark this answers is not in the document]
Comment (rishumehra) [reply modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc:15]: Thanks for pointing this.

File: modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
++
+.Searching Console resource
```
[the remark this answers is not in the document]
Comment (rishumehra) [reply modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc:17]: Yes, this is a title for the image. I've updated the preview link in the description section please check.

File: modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+image::odc_cluster_console.png[]
+. Select *cluster* under *Name* list.
```
[the remark this answers is not in the document]
Comment (rishumehra) [reply modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc:19]: Done

File: modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc (hunk 6 above)
Comment (rishumehra) [reply modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc:21]: That's a good point too. I'll try to rephrase.

**Review by rishumehra — COMMENTED** [review]
@missmesss Kindly check the comments from me, we need to discuss these. Feel free to leave a reply on them or discuss them in the meeting scheduled for today. Thanks!

File: modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc (hunk 7 above)
[the remark this answers is not in the document]
Comment (missmesss) [reply modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc:8]: @rishumehra I'm afraid I still don't quite understand why this "Display name" feature is mentioned in the introduction. If I understand @jerolimov 's comment correctly, it's an extra fine-tuning option. Does it help the users to read about some unnecessary option as the first thing in the module?

File: modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc (hunk 10 above)
Comment (missmesss) [inline modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc:15]: @rishumehra Now that you've provided the preview of a rendered doc with the screenshots, I'm afraid I got confused by this line a bit.
It looks like the step suggests adding a new row that will be "Console `operator.openshift.io/v1". Am I misreading it? Does my suggestion make sense?

```suggestion
. Select *Console `operator.openshift.io/v1`* in the Resources list (to add a new role? Is this tail needed?)
```

**Comment by missmesss** [comment]
@rishumehra Thank you for looking at the comments attentively and implementing some changes.

Of course, we should all follow our products docs guidelines, sorry, I'm not well-aware of OpenShift docs best practices and suggested some things based on my previous experiences.

I got 2 more questions, please (for lines 8 and 15). Please, let me know what you think.

File: modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc (hunk 7 above)
[the remark this answers is not in the document]
Comment (rishumehra) [reply modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc:8]: @missmesss Got it, so I'll put in something more appropriate like "v1" or something else to avoid such confusion.

File: modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc (hunk 10 above)
Comment (rishumehra) [reply modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc:15]: @missmesss Let me demo it to you for this one, then we can discuss this further. Because now I am also not sure about it.

Comment (missmesss) [reply modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc:15]: @rishumehra Do you mean some video demo or presentation? We could save you some time as anyway I don't need to know how it works in detail. It'll be great if you check that this line is accurate, that'll be enough, thank you.

File: modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc (hunk 7 above)
[the remark this answers is not in the document]
Comment (missmesss) [reply modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc:8]: @rishumehra But do we need such details in the introduction at all? In some other comment you stated we needed an action item in the introduction to the procedure, where does this guideline come from, please?

Comment (missmesss) [inline modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc:8]: To sum up all my confusions: the current introduction includes a tip about displaying access roles names, it doesn't describe what the procedure is about.

Also, customize and configure are basically the same here, are they not?

```suggestion
You can customize cluster roles of existing projects by editing the YAML code.
```

File: modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc (hunk 10 above)
Comment (rishumehra) [reply modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc:15]: @missmesss 

Here is a procedure overview for your reference:

- Actually in *Resource* tab we search for "Console" and select the "Console `operator.openshift.io/v1" resource. Like this: 
  image

- Then add ""Console `operator.openshift.io/v1"" item to a resource list, like this:
 image

- Then inside it we edit the YAML code to edit the list of user access roles.
  image

Comment (rishumehra) [reply modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc:15]: To simplify I'll just keep: 

. Select the **Console `operator.openshift.io/v1`**.

File: modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc [hunk 13]
```diff
@@ -0,0 +1,39 @@
+// Module included in the following assemblies:
+//
+// applications/projects/working-with-projects.adoc
+
+[id="odc-customizing-available-cluster-roles-using-developer-perspective_{context}"]
+= Customizing the available cluster roles using the Developer perspective
+
+You can customize cluster role's name of existing projects by editing the YAML code in the `console.openshift.io/display-name: v1`.
```
Comment (jc-berger) [inline modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc:8]: Some missing articles (see my rephrase). Also, what's the point of customizing the cluster role's name? See my rephrase to give a very quick, simple explanation of why users would care to customize the cluster role's name:

`Customize the cluster role's name of existing projects by editing the YAML code in the `console.openshift.io/display-name: v1`. When you customize the cluster role's name, you can <state benefit>.`

File: modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc [hunk 14]
```diff
@@ -0,0 +1,39 @@
...
+
+.Procedure
+Customize the cluster role options of all projects that are shown in the *Project Access* tab:
```
Comment (jc-berger) [inline modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc:11]: Above, you say to customize the cluster role's name. Here, you say to customize the cluster role's options. Which is it? The title says "roles" so perhaps here you can say "role" instead of "option" for consistency? Unless they are two different things?

Also, you sometimes use possessive: cluster role's name. 
Other times, you don't use a possessive: cluster role options. 

Pick one to be consistent. I would go with whichever case is more common in openshift docs. If it helps, grammatically, it would be possessive since both the name and options belong to the cluster role.

File: modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc [hunk 15]
```diff
@@ -0,0 +1,39 @@
...
+
+. In the *Search* view, select the *Resources* drop down and search "*Console*".
```
Comment (jc-berger) [inline modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc:13]: Your previous sentence says that we want to be in the *Project Access* tab. In your first step, you tell me to go to the *Search* view -> *Resources* -> *Console* So am I supposed to be in the Console or Project Access tab? Or is the Console within the Project Access tab?

Honestly, I'm not even sure if you need the first sentence before the procedure, especially since it can cause confusion as to where to direct the user. I think if you make the use case clearer in the abstract, you can simply start with step 1 under the procedure. So: 

**Title**
<abstract>
**Procedure**
1....

File: modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc [hunk 16]
```diff
@@ -0,0 +1,39 @@
...
+. Select the *Console `operator.openshift.io/v1`*.
++
+.Searching Console resource
+image::odc_cluster_console.png[]
```
Comment (jc-berger) [inline modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc:17]: Though I find the screenshot helpful, I know they can be quite difficult for accessibility, a growing concern in RH documentation. They also are hard to maintain up to date. 

Personally, I find it best to avoid them. However, if your team for this specific document believes the screenshot would help all users, keep it.

File: modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc [hunk 17]
```diff
@@ -0,0 +1,39 @@
...
+. Select *cluster* under the *Name* list.
```
Comment (jc-berger) [inline modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc:18]: Verify that cluster isn't capitalized. In the screenshot, I see all the UI elements are capitalized, so "cluster" here very well may need to be capitalized as well.

File: modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc [hunk 18]
```diff
@@ -0,0 +1,39 @@
...
+. Navigate to *YAML* tab to view and edit the YAML code.
```
Comment (jc-berger) [inline modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc:19]: Navigate to _the_ *YAML* tab..

File: modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc (hunk 16 above)
Comment (jc-berger) [reply modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc:17]: If you do keep the screenshot, end step two with a colon, not a period, since it introduces the screenshot:

2. Select the Console operator:

File: modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc [hunk 19]
```diff
@@ -0,0 +1,39 @@
...
+. In the YAML code under `spec` add or edit the list of `availableClusterRoles`.
```
[the remark this answers is not in the document]
Comment (jc-berger) [reply modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc:20]: Since this step proceeds the yaml snippet, end it with a colon.

File: modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc [hunk 20]
```diff
@@ -0,0 +1,39 @@
...
+  customization:
+    projectAccess:
+      availableClusterRoles:
+      - admin
+      - edit
+      - view
+----
++
+For example, you can remove `view` role from the list to show `admin` and `edit` roles only.
+. Click *Save*.
```
Comment (jc-berger) [inline modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc:34]: I think this step could be combined with the previous step. Something like this: 

`In the YAML code under `spec`, add or edit the list of `availableClusterRoles` and save your changes:
YAML snippet.

File: modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc [hunk 21]
```diff
@@ -0,0 +1,39 @@
...
+
+Verify the changes in the *Role* drop down under the *Project Access* tab of *Project* view.
```
Comment (jc-berger) [inline modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc:36]: Make this a verification section:

.Verification 

- To verify your changes, go to the *Project* view > *Project Access* Tab > *Role* drop-down and confirm your changes have been made. 


Also, with these instructions being fairly simple, I don't think the screenshot is necessary. Granted, I'm not the biggest fan of screenshots, so I might be biased :)

File: modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc [hunk 22]
```diff
@@ -0,0 +1,39 @@
...
+
+.Project roles
```
Comment (jc-berger) [inline modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc:38]: I don't see the project roles within this procedure in the preview link.

File: modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc (hunk 14 above)
Comment (rishumehra) [reply modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc:11]: Good point, I'll keep this suggestion while rephrasing the content.

File: modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc (hunk 15 above)
Comment (rishumehra) [reply modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc:13]: Sounds good. Will try to simplify it a bit more.

File: modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc (hunk 16 above)
Comment (rishumehra) [reply modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc:17]: In ODC we keep the sentence closed with a period. For reference check this page: https://docs.openshift.com/container-platform/4.9/applications/odc-viewing-application-composition-using-topology-view.html#odc-interacting-with-applications-and-components_viewing-application-composition-using-topology-view

File: modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc (hunk 18 above)
Comment (rishumehra) [reply modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc:19]: Done.

File: modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc (hunk 17 above)
Comment (rishumehra) [reply modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc:18]: I'll verify it again.

File: modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc (hunk 19 above)
[the remark this answers is not in the document]
Comment (rishumehra) [reply modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc:20]: Sure.

File: modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc (hunk 20 above)
Comment (rishumehra) [reply modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc:34]: Sounds like a good suggestion. I'll try this one too.

File: modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc (hunk 21 above)
Comment (rishumehra) [reply modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc:36]: Thanks for the suggestion. I'll check whether do we keep `.Verficiation` step in ODC or not as I never encountered one so far. For the screenshot I might wanna keep it as it helps users to understand things better.

File: modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc (hunk 22 above)
Comment (rishumehra) [reply modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc:38]: Those are there in the screenshot. Let us check again.

**Review by rishumehra — COMMENTED** [review]
@jc-berger Thanks for your comments and feedback. I replied to them and will make the changes accordingly.

File: modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc (hunk 13 above)
Comment (rishumehra) [reply modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc:8]: The first part of the suggestion aligns with what is expected on the page. The other part is not required. So I'll rephrase accordingly.

File: modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc (hunk 17 above)
Comment (rishumehra) [reply modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc:21]: The "cluster" is not capitalized in the UI. Check this:

image

File: modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc [hunk 23]
```diff
@@ -0,0 +1,34 @@
+// Module included in the following assemblies:
+//
+// applications/projects/working-with-projects.adoc
+
+[id="odc-customizing-available-cluster-roles-using-developer-perspective_{context}"]
+= Customizing the available cluster roles using the Developer perspective
+
+Customize the cluster role options of an existing project mentioned in the *Project* > *Project access* tab > *Role* drop down, by editing the YAML code in the `console.openshift.io/display-name: v1`.
```
Comment (jc-berger) [inline modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc:8]: This abstract seems more like a procedure step to me. 

How about giving a one sentence overview of why they'd customize the cluster role options? Something like:

`To have more control over your projects and to make changes to your project based off your needs, customize the cluster role options of an existing project.` 

I'd then scratch this last sentence from the abstract and make it step one of the procedure. 

1. Select  *Project* > *Project access* tab > *Role* drop down and edit the  YAML code in the `console.openshift.io/display-name: v1`.
2. Search for "*Console*" in the *Search* view > *Resources* drop down.
3. So on and so forth.

Comment (rishumehra) [reply modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc:8]: Sounds fair. Let me rephrase a bit more based on your example.

File: modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc [hunk 24]
```diff
@@ -0,0 +1,34 @@
+// Module included in the following assemblies:
+//
+// applications/projects/working-with-projects.adoc
+
+[id="odc-customizing-available-cluster-roles-using-developer-perspective_{context}"]
+= Customizing the available cluster roles using the Developer perspective
+
+Each user of a project is assigned to a cluster role based on their access control. These cluster roles are mentioned in the *Project* view > *Project access* tab > *Role* drop down. By default these roles are *Admin*, *Edit*, and *View*. To add or edit the cluster roles for a project, you can customize the YAML code of the cluster.
```
Comment (jc-berger) [inline modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc:8]: See grammatical changes: 

The users of a project are assigned to a cluster role based on their access control. These cluster roles are mentioned in the *Project* view > *Project access* tab > *Role* drop down. By default, these roles are *Admin*, *Edit*, and *View*. To add or edit the cluster roles for a project, you can customize the YAML code of the cluster.

**Review by jc-berger — APPROVED** [review]
/lgtm

**Review by jc-berger — APPROVED** [review]
/lgtm

File: modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc [hunk 25]
```diff
@@ -0,0 +1,34 @@
...
+
+The users of a project are assigned to a cluster role based on their access control. These cluster roles are mentioned in the *Project* view > *Project access* tab > *Role* drop down. By default, these roles are *Admin*, *Edit*, and *View*. To add or edit the cluster roles for a project, you can customize the YAML code of the cluster.
+
+.Project roles
+image::odc_cluster_role.png[]
+
+.Procedure
+To customize the different cluster roles of a project:
+
+. Search for "Console" in the *Search* view > *Resources* drop down.
```
Comment (Preeticp) [inline modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc:16]: When writing for GUI, location then action.
Suggest: In the *Search* view, use the *Resources* drop-down list to search for `Console`.
User defined items use backticks. We never use quotation marks in our docs. Please see the Asciidoc markup section in the OCP guidelines on GH: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#quick-markup-reference

File: modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc [hunk 26]
```diff
@@ -0,0 +1,34 @@
...
+. Select the *Console `operator.openshift.io/v1`*.
```
Comment (Preeticp) [inline modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc:17]: Suggest: From the available options, select...

File: modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc [hunk 27]
```diff
@@ -0,0 +1,34 @@
...
+// applications/projects/working-with-projects.adoc
+
+[id="odc-customizing-available-cluster-roles-using-developer-perspective_{context}"]
+= Customizing the available cluster roles using the Developer perspective
+
+The users of a project are assigned to a cluster role based on their access control. These cluster roles are mentioned in the *Project* view > *Project access* tab > *Role* drop down. By default, these roles are *Admin*, *Edit*, and *View*. To add or edit the cluster roles for a project, you can customize the YAML code of the cluster.
+
+.Project roles
+image::odc_cluster_role.png[]
+
```
Comment (Preeticp) [inline modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc:12]: Suggest dropping this screenshot from maintenance pov. The instructions are clear enough. Also, the previous module has a similar screenshot, though that might not be relevant when we consider each module to be self sufficient. But like I said the instructions are clear in this case.

File: modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc (hunk 17 above)
Comment (Preeticp) [reply modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc:21]: Principle of location before action: Suggest: Under the *Name* list, select...
The user needs to know where the field is before he/she takes action on it.

**Review by Preeticp — COMMENTED** [review]
@rishumehra overall looks good. I have added a few comments, PTAL.

File: modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc (hunk 14 above)
Comment (missmesss) [reply modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc]: @jc-berger Thank you, Jake, for looking into this.
It was one of my concerns too: the introductory paragraph seems a bit confusing to me since it describes some fine-tuning option (customizing roles names) while it's not this module's main topic. I hope you could help us solve this, thanks.

File: modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc (hunk 27 above)
Comment (rishumehra) [reply modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc:12]: Agreed with your point on this. Let me remove the screenshot from here.

File: modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc (hunk 26 above)
Comment (rishumehra) [reply modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc:17]: Done

File: modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc [hunk 28]
```diff
@@ -0,0 +1,34 @@
+// Module included in the following assemblies:
+//
+// applications/projects/working-with-projects.adoc
+
+[id="odc-customizing-available-cluster-roles-using-developer-perspective_{context}"]
+= Customizing the available cluster roles using the Developer perspective
+
+The users of a project are assigned to a cluster role based on their access control. These cluster roles are mentioned in the *Project* view > *Project access* tab > *Role* drop down. By default, these roles are *Admin*, *Edit*, and *View*. To add or edit the cluster roles for a project, you can customize the YAML code of the cluster.
```
[the remark this answers is not in the document]
Comment (rishumehra) [reply modules/odc-customizing-available-cluster-roles-using-developer-perspective.adoc:8]: Done

**Comment by pneedle-rh** [comment]
Note: I am updating the section that was added in this PR to use the Administrator perspective and to clarify that this changes the available roles for all projects. I am also adding a prerequisite about needing cluster admin privileges. I tested the procedure in the Developer perspective without cluster admin privileges and got the following error:

```
Error details
consoles.operator.openshift.io is forbidden: User "<username>" cannot list resource "consoles" in API group "operator.openshift.io" at the cluster scope
```
